crates/admin/src/cluster_controller/service/scheduler.rs

```diff
@@ -236,20 +251,69 @@ impl<T: TransportConnect> Scheduler<T> {
         //  they learn about the updated nodes configuration. To reduce the risk of this, we should
         //  wait a little bit to give the nodes configuration time to be spread across the cluster.
 
-        self.instruct_nodes(legacy_cluster_state)?;
+        Ok(())
+    }
+
+    async fn load_all_partition_configuration(
+        &mut self,
+        partition_table: &PartitionTable,
+    ) -> Result<(), Error> {
+        if !self.partitions.is_empty() {
+            return Ok(());
+        }
+
+        self.partitions = futures::stream::iter(partition_table.iter_ids().cloned().map(
+            async |partition_id| {
+                Result::<_, Error>::Ok((
+                    partition_id,
+                    Self::load_partition_configuration(
+                        self.metadata_writer.raw_metadata_store_client(),
+                        partition_id,
+                    )
+                    .await?,
+                ))
+            },
+        ))
+        .buffer_unordered(16) // load partitions concurrently
+        .try_filter_map(
+            async |(partition_id, partition_state)| match partition_state {
+                Some(partition_state) => {
+                    Self::note_observed_membership_update(
```

Comment:
unsure if this is needed. it would be set when upserting initial config, even if that initial config hasnt changed from what was already in metadata. in this code path we are only loading whats in metadata and not doing any mutation. my thinking is it doesn't hurt to update the observed membership.

Reply:
I think this is actually required because otherwise we will only update the current and next configuration on an actual reconfiguration.